### PR TITLE
Friendships v1 

### PR DESCRIPTION
**Pull request by petudeveloper**
Create a model with associations and all requested features for friendships.

### CODE REVIEW

**Review by BrianSammit — CHANGES_REQUESTED** [review]
STATUS:INVALID :yellow_circle: 
------------
Hello @petumazo @YuvrajM108, this pull request will be marked as invalid because the Linters are not showing on GitHub, Please fix this and send it again.

**Github Actions**

If the student's GitHub Action linters are not being triggered properly, ask the students to follow the steps below on the forked repo:

* Enable the GitHub Actions in the Actions tab.
* Create the feature/branch and push.
* Start working on the milestone as usual.
* Open a PR from the feature/branch when the work is done.

These same instructions are available in the forked [ror-social-media repository README](https://github.com/microverseinc/ror-social-scaffold#github-actions).

**Comment by petudeveloper** [comment]
@BrianSammit Thanks for the feedback. We followed your suggested steps, thank you.
Have a good day. 👋

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: :stop_sign: 
-------------
Hello again team :wave:, Great job you are doing in this milestone :muscle: :clap: 
TO HIGHLIGHT: 

- Your project is professional 
- Your project is professional 
- The user model is well structured
- The friendship model is well structured 
- Well structured files

However,  there are a few changes to make to complete this milestone :wrench: see them below :point_down: 
----------

- [x] Posts timeline includes posts written by logged-in user and their friends. Currently, I can see posts from no friends. Please fix this. 
 

friends_timeline

- [x] Please add unit and Integration tests! 

tests

---------------
Make these minor changes and send it again for another review :+1: :muscle: 

HAPPY CODING !!!
SLACK: Brian Sammit Cruz Rodriguez

MB5W4D

**Comment by petudeveloper** [comment]
@BrianSammit Thanks for the feedback. Following your suggestions we:

1. Modify posts timeline to include posts written by logged-in user and their friends. 
2. Add unit and Integration tests.

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coders, 👋🏽 
**This is a great project you have going on, well-done lads and thanks for implementing the changes requested earlier on**. 

### Well-done 💯 
    1. The app runs just fine.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] Posts timeline includes posts written by logged-in user and their friends. Currently, I can see posts from no friends. This issue is still present and not entirely solved. Look into it.
- [x] You have 5/12 tests failing. Give them a fix.
Screenshot from 2021-07-20 05-19-37

- [x] Add unit tests for all your models.

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      expect(post.valid?).to be(false)
+    end
+  end
+
+  describe 'Post associations' do
+    it 'deletes dependent objects' do
+      user = User.find_by(name: 'User1')
+      expect(user.class).to be(User)
+      post = user.posts.create(content: 'A simple post')
+      expect(post.likes.count).to eq(0)
```
Comment (bini-i) [inline spec/models/post_spec.rb:22]: - Kindly make sure to have one expect per it block as it helps you be specific about what you are testing. [**optional**]

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes Required :yellow_circle:            
### Hello @petumazo  :wave: 

Thanks for working on the requirements suggested by the previous reviewers. Your project looks good and shows that you have a lot of effort into it :clap: . Accepting and rejecting requests works flawlessly, integration tests are included, and straightforward project setup are some of the highlights :100: .

#### To improve
- kindly add unit tests for public methods in your user model, currently, you only have unit tests for post validation and association. [**Required**]

#### If you have any questions feel free to ask within the PR and ping me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (Code Reviewer)
> GitHub: [@bini-i](https://github.com/bini-i) 
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)#

**Comment by petudeveloper** [comment]
@bini-i Thanks for the feedback. We followed your suggestions and made the following corrections:

- Add unit tests for public methods in the user model.

**Review by bellom — APPROVED** [review]
## Hi @petumazo  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best**

**Pull request merged**